code/datums/components/fantasy/suffixes.dm

```diff
@@ -189,22 +195,79 @@
 	name = "curse of hunger"
 	placement = AFFIX_SUFFIX
 	alignment = AFFIX_EVIL
+	weight = 5
 
 /datum/fantasy_affix/curse_of_hunger/validate(obj/item/attached)
-	//curse of hunger that attaches onto food has the ability to eat itself. it's hilarious.
-	if(!IS_EDIBLE(attached))
-		return TRUE
+	// Curse of hunger can be really unbearable to deal with,
+	// so it should not start on someone or in a bag.
+	if(!isturf(attached.loc))
+		return FALSE
 	return TRUE
 
 /datum/fantasy_affix/curse_of_hunger/apply(datum/component/fantasy/comp, newName)
 	. = ..()
 	var/obj/item/master = comp.parent
 	var/filter_color = "#8a0c0ca1" //clarified args
 	var/new_name = pick(", eternally hungry", " of the glutton", " cursed with hunger", ", consumer of all", " of the feast")
-	master.AddElement(/datum/element/curse_announcement, "[master] is cursed with the curse of hunger!", filter_color, new_name, comp)
+	master.AddElement(/datum/element/curse_announcement, "[master] is cursed with the curse of polymorph!", filter_color, "", comp)
 	comp.appliedComponents += master.AddComponent(/datum/component/curse_of_hunger)
-	return newName //no spoilers!
+	return "[newName][new_name]" //no spoilers!
 
 /datum/fantasy_affix/curse_of_hunger/remove(datum/component/fantasy/comp)
 	var/obj/item/master = comp.parent
 	master.RemoveElement(/datum/element/curse_announcement) //just in case
+
+/datum/fantasy_affix/curse_of_polymorph
+	name = "curse of polymorph"
+	placement = AFFIX_SUFFIX
+	alignment = AFFIX_EVIL
+
+/datum/fantasy_affix/curse_of_polymorph/validate(obj/item/attached)
+	// Curse of hunger can be really unbearable to deal with,
```

Comment:
`/datum/fantasy_affix/curse_of_polymorph`
`// Curse of hunger can be really unbearable to deal with`